.gitignore

```diff
@@ -333,3 +333,8 @@ ASALocalRun/
 
 # MFractors (Xamarin productivity tool) working folder 
 .mfractor/
+
+# Adding the debug files automatically generated in VS Code
+tasks.json
+launch.json
```

Comment:
Including these files is intentional. They need to be there or people cannot debug/run the examples out of the box or build the code from the task commands. Even though most of that content is auto generated, it is auto-generated incorrectly. Here is what the auto generated looks like in `launch.json`:
```json
{
	"name": ".NET Core Launch (console)",
	"type": "coreclr",
	"request": "launch",
	"preLaunchTask": "build",
	"program": "${workspaceFolder}/Examples/BasicsAndExtensions/bin/Debug/net5.0/BasicsAndExtensions.dll",
	"args": [],
	"cwd": "${workspaceFolder}/Examples/BasicsAndExtensions",
	"console": "internalConsole",
	"stopAtEntry": false
},
```
Problems:
- the `name` is bad. it is a general `".NET Core Launch (Console)"` rather than telling you what actual project you are running
- the `preLaunchTask` is again a generalized name that is not relative to the project
- the `console` is bad. It needs to be `"externalTerminal"` rather than `"internalConsole"`. The internal view in VSC _does not support console input_.

Reply:
Did you buy chance open a folder other than the root folder of the repo? If you open the root folder VSC should not auto generate any other `json` files.

Reply:
> Did you buy chance open a folder other than the root folder of the repo? If you open the root folder VSC should not auto generate any other `json` files.

Exactly. I think we went over [this](https://github.com/ZacharyPatten/Towel/pull/79#issuecomment-851793118). I kinda forgot about the specific json files you wanted to preserve. Should I rather place a `.gitignore` file to ignore these inside the folder as well?

Reply:
Gotcha. I understand now. I just want to make sure the `.vscode` folder in the root is maintained, but we can `.gitignore` all the other ones. VSC os adding the `tasks.json` and `launch.json` inside of a `.vscode` folder for you right? We can probably just `.gitignore` all `.vscode` folders that are not in the root.

Reply:
I've made the change in this commit